pkg/diagnostics/networkpod/node.go

```diff
@@ -0,0 +1,79 @@
+package network
+
+import (
+	"errors"
+	"fmt"
+
+	kapi "k8s.io/kubernetes/pkg/api"
+	kclient "k8s.io/kubernetes/pkg/client/unversioned"
+	kexec "k8s.io/kubernetes/pkg/util/exec"
+
+	"github.com/openshift/origin/pkg/diagnostics/networkpod/util"
+	"github.com/openshift/origin/pkg/diagnostics/types"
+	"github.com/openshift/origin/pkg/util/netutils"
+)
+
+const (
+	CheckNodeNetworkName = "CheckNodeNetwork"
+)
+
+// CheckNodeNetwork is a Diagnostic to check that all nodes in the cluster are accessible within a pod
+type CheckNodeNetwork struct {
+	KubeClient *kclient.Client
+}
+
+// Name is part of the Diagnostic interface and just returns name.
+func (d CheckNodeNetwork) Name() string {
+	return CheckNodeNetworkName
+}
+
+// Description is part of the Diagnostic interface and just returns the diagnostic description.
+func (d CheckNodeNetwork) Description() string {
+	return "Check that all nodes in the cluster are accessible within a pod"
+}
+
+// CanRun is part of the Diagnostic interface; it determines if the conditions are right to run this diagnostic.
+func (d CheckNodeNetwork) CanRun() (bool, error) {
+	if d.KubeClient == nil {
+		return false, errors.New("must have kube client")
+	}
+	return true, nil
+}
+
+// Check is part of the Diagnostic interface; it runs the actual diagnostic logic
+func (d CheckNodeNetwork) Check() types.DiagnosticResult {
+	r := types.NewDiagnosticResult(CheckNodeNetworkName)
+
+	nodes, err := util.GetNodes(d.KubeClient)
+	if err != nil {
+		r.Error("DNodeNet1001", err, err.Error())
+		return r
+	}
+
+	for _, node := range nodes {
+		checkNodeConnection(&node, r)
```

Comment:
Good point. I will change the test to verify if the pod can access its own node.